pySNOM/images.py

```diff
@@ -1,118 +1,121 @@
 import numpy as np
 import copy
 from enum import Enum
-import pySNOM.defaults as defaults
+from pySNOM.defaults import Defaults
+from gwyfile.objects import GwyDataField
 
 MeasurementModes = Enum('MeasurementModes', ['None','AFM', 'PsHet', 'WLI', 'PTE', 'TappingAFMIR', 'ContactAFM'])
 DataTypes = Enum('DataTypes', ['Amplitude', 'Phase', 'Topography'])
 ChannelTypes = Enum('ChannelTypes', ['Optical','Mechanical'])
 
 # Full measurement data containing all the measurement channels
-class Data:
-    def __init__(self, filename = None, data = None, parameters = None, mode = "None"):
-        self.filename = filename  # Full path with name
-        # Measurement mode (PTE, PSHet, AFM, NanoFTIR) - Enum MeasurementModes
-        self.mode = MeasurementModes[mode]
-        # Full data from the gwy files with all the channels
-        self.data = data # All channels - Dictionary
-        # Other parameters from info txt -  Dictionary
-        if parameters is not None:
-            self.setParameters(parameters)
-        else:
-            self.parameters = parameters
-
-    # Method to set the actual data read from the gwyddion or gsf files
-    def setData(self, data: dict):
-        self.data = data
-
-    # Method to set measurement mode directly
-    def setMeasurementMode(self, measmode: str):
-        self.mode = MeasurementModes[measmode]
-
-    # Method to retrieve e specific channel from all the measurement channels
-    def getChannelData(self, channelname: str):
+class Measurement:
+    def __init__(self, filename = None, data = None, info = None, mode = "None"):
```

Comment:
Why does this class both `filename` and `data` arguments?

I'd say that readers need filenames, and actually read data only needs data. Unless this classes are not meant to do anything with the filename because it acts only as annotation. In this case, `data` should go first.

Per PEP-8 we do not make spaces around `=` in arguments lists.

Reply:
I'd also make `data` non-optional argument for this one.

Reply:
Yes, sure that is true. `filename` is only for annotation. I make the changes, thanks for the suggestions.